### PR TITLE
Notifications: Update styling and switch to React class

### PR DESCRIPTION
**Pull request by dmsnell**
This patch updates the styling and formatting for the file which
interacts with the notifications iframe.

This file shas been confusing to work in and it needs work but
improvements are difficult to push in with all the noise from the style.

This patch thus is intended to update the file without making behavioral
changes in order to prepare it for actual functional updates.

**Testing**

Please smoke test the various ways of interacting with the notes panel.

 - The global `n` hotkey should toggle its display
 - Resize Calypso while the panel is open
 - Navigate around notes

I have already performed a variety of tests to try and confirm that it's working.

### CODE REVIEW

File: client/notifications/index.jsx [hunk 1]
```diff
@@ -1,58 +1,85 @@
...
  * Internal dependencies
  */
-var analytics = require( 'lib/analytics' ),
-	config = require( 'config' ),
-	user = require( 'lib/user' )();
+import analytics from 'lib/analytics';
+import config from 'config';
+import userLib from 'lib/user';
+
+const user = userLib();
```
Comment (gwwar) [inline client/notifications/index.jsx]: Optional, but this can go under module variables

File: client/notifications/index.jsx [hunk 2]
```diff
@@ -206,80 +259,101 @@ var Notifications = React.createClass( {
...
+			// we deploy to beta before pushing to production
+			config.isEnabled( 'notifications2beta' ) && 'beta',
+			'/',
+			// default app is at index.html
+			// auto-rtl is used to generate
+			// the flipped copy
+			user.isRTL() && 'rtl.html',
+			`?locale=${ localeSlug }`,
+			// bust cache every hour
+			// @TODO replace this with a strong process that doesn't make users download so much data
```
Comment (gwwar) [inline client/notifications/index.jsx]: In the future I would recommend using the same `git describe` output from the notifications-client repo

File: client/notifications/index.jsx [hunk 3]
```diff
@@ -206,80 +259,101 @@ var Notifications = React.createClass( {
...
+			`&cache_buster=${ now.getFullYear() }${ now.getMonth() + 1 }${ now.getDate() }${ now.getHours() }`,
+		].filter( identity ).join( '' );
```
Comment (gwwar) [inline client/notifications/index.jsx]: Oh never thought of using identity there and depending on the truthyness of each item.

Optional, but also consider: `array.filter( Boolean ).join( '' )` or `compact( array ).join( '' )`.

File: client/notifications/index.jsx (hunk 2 above)
Comment (dmsnell) [inline client/notifications/index.jsx]: In the future I will! :smile:

File: client/notifications/index.jsx (hunk 3 above)
Comment (dmsnell) [inline client/notifications/index.jsx]: `compact()` is probably the better idiom here. thanks. I often forget about this.

Comment (dmsnell) [inline client/notifications/index.jsx]: used `compact()` in 13ca010

File: client/notifications/index.jsx (hunk 1 above)
Comment (dmsnell) [inline client/notifications/index.jsx]: moved in 13ca010

File: client/notifications/index.jsx [hunk 4]
```diff
@@ -123,70 +151,95 @@ var Notifications = React.createClass( {
...
-			const token = oAuthToken.getToken();
+		const token = oAuthToken.getToken();
 
-			if ( token !== false ) {
-				this.postMessage( {
-					action: 'setAuthToken',
-					token: token
-				} );
-			}
+		if ( token === false ) {
```
Comment (rodrigoi) [inline client/notifications/index.jsx]: any particular reason not use just `( ! token )` ?

Comment (dmsnell) [inline client/notifications/index.jsx]: just because I am trying not to change the behaviors here and it was using strict equality before

**Pull request merged**